packages/@sanity/migrate/src/runner/normalizeMigrateDefinition.ts

```diff
@@ -58,30 +103,48 @@ async function collectDocumentMutations(
   migration: NodeMigration,
   doc: SanityDocument,
   context: MigrationContext,
-) {
+): Promise<(Mutation | Transaction)[]> {
   const documentMutations = Promise.resolve(migration.document?.(doc, context))
   const nodeMigrations = flatMapDeep(doc as JsonValue, async (value, path) => {
     const [nodeReturnValues, nodeTypeReturnValues] = await Promise.all([
       Promise.resolve(migration.node?.(value, path, context)),
       Promise.resolve(migrateNodeType(migration, value, path, context)),
     ])
 
-    return [...arrify(nodeReturnValues), ...arrify(nodeTypeReturnValues)].map((change) =>
-      normalizeNodeMutation(path, change),
+    return [...arrify(nodeReturnValues), ...arrify(nodeTypeReturnValues)].map(
+      (change) => change && normalizeNodeMutation(path, change),
     )
   })
 
   return (await Promise.all([...arrify(await documentMutations), ...nodeMigrations]))
     .flat()
-    .map((change) => normalizeDocumentMutation(doc._id, change))
+    .flatMap((change) => (change ? normalizeDocumentMutation(doc._id, change) : []))
+    .flat()
```

Comment:
❓ is the two `.flat()` correct? Could it just flatten it just at the end?

Reply:
You're absolutely right. Removed the last one. Thanks!